backend_kvm.go

```diff
@@ -0,0 +1,136 @@
+// +build linux
+// +build amd64
+
+package fakemachine
+
+import (
+	"fmt"
+	"os"
+	"strings"
+)
+
+type KvmBackend struct {}
+
+func init() {
+	RegisterBackend(KvmBackend{})
+}
+
+func (b KvmBackend) Name() string {
+	return "kvm"
+}
+
+func (b KvmBackend) Supported() (bool, error) {
+	if _, err := os.Stat("/dev/kvm"); err != nil {
+		return false, err
+	}
+	return true, nil
+}
+
+func (b KvmBackend) InitScriptModules() []string {
+	return []string{"virtio_pci", "virtio_console", "9pnet_virtio", "9p"}
+}
+
+func (b KvmBackend) InitScriptExtraCommands() []string {
+	return nil
+}
+
+func (b KvmBackend) NetworkdMatch() string {
+	return "e*"
+}
+
+func (b KvmBackend) CmdMountVolume(mount mountPoint) string {
+	return fmt.Sprintf("busybox mount -v -t 9p -o trans=virtio,version=9p2000.L,cache=loose,msize=262144 %s %s", mount.label, mount.machineDirectory)
+}
+
+func (b KvmBackend) GetFstabEntry(mount mountPoint) string {
+	return fmt.Sprintf("%s %s 9p trans=virtio,version=9p2000.L,cache=loose,msize=262144 0 0", mount.label, mount.machineDirectory)
```

Comment:
And on top of the CmdMountVolume comments; there is obvious pattern here ;)  You could have one interface function which returns (<fstype>, <fsoptions>) and from that you can work out both the fstab entry *and* the mount command to run in busybox.

Reply:
ack, think we come to the same conclusion  ;-)